contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java

```diff
@@ -834,17 +840,16 @@ private boolean isPartitionRequired(int opid, List<KafkaConsumer.KafkaMeterStats
         return false;
       }
 
+      if (!isPartitionBasedOnLoad) {
```

Comment:
Consider the below scenario:
By default, repartitionInterval is 30 sec. So, operator checks whether the repartition is needed or not at 30, 60, 90, 120, etc. Suppose, Meta-data change happened at 120.
In this case, operator checks whether repartition required or not at 30, 60, 90 secs. isPartitionRequired returns true based on HardConstraint and SoftConstraint.
But, if the user wants to disable the dynamic partition based on load then isPartitionRequied() has to return false at 30, 60, 90 secs.